### PR TITLE
Sierra: Fix and improve item/bib handling with transaction history

### PR DESCRIPTION
**Pull request by EreMaijala**
- Allows a bib to be fetched also when the item has already been deleted
- Fixes an issue when the transaction list contained multiple items for the same bib where only one item had it's bib mapped to it.

### CODE REVIEW

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php [hunk 1]
```diff
@@ -2366,18 +2365,6 @@ protected function getItemStatusesForBib(string $id, bool $checkHoldings, ?array
...
     }
 
-    /**
-     * Extract the actual call number from item's call number field
-     *
-     * @param string $callNumber Call number field
-     *
-     * @return string
-     */
-    protected function extractCallNumber(string $callNumber): string
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: This seems to be reverting changes from another recent Sierra PR. Is that intentional? It doesn't seem related to the intent of this one...

**Review by demiankatz — CHANGES_REQUESTED** [review]
One thing caught my eye right away, and I thought I'd better ask about it before looking at the rest of the code in more detail.

File: module/VuFind/src/VuFind/ILS/Driver/SierraRest.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/ILS/Driver/SierraRest.php]: Oops, I thought I took care of that. That's what you get for working with so many different versions. Should be fixed now.

**Review by demiankatz — APPROVED** [review]
Looks good to me -- thanks, @EreMaijala!
